File: 03-frontend-booking-system/booking-system-web/src/app/features/slots/dtos/slot-response-dto.ts
import { TimeRange } from "@shared/model/time-range";

export class SlotResponseDto {
    eventId!:number;
    id!: number;
    slotName!: string;
    slotDescription?:string;
    slotStartTime!: string;
    slotEndTime!: string;
    maxBookingsPerIdentity!: number | null;
    maxBookPerInterval!: number;
    slotIntervalMinutes!: number;
    slotFrequencyIntervalMinutes?: number;
    createdAt!: string;
    updatedAt!: string;
    businessDaysHours?: Record<number,TimeRange[]>;
    businessTimeZone?: string;
    businessAllowOt?: boolean;
    flexibleDaysHours?: TimeRange[];
    bookingsCount!: number;
}

File: 02-backend-booking-system/src/main/java/com/littlek4za/booking_system/features/slot/dto/SlotResponseDto.java
package com.littlek4za.booking_system.features.slot.dto;

import java.time.Instant;
import java.util.List;
import java.util.Map;

import com.littlek4za.booking_system.features.slot.model.InstantRange;
import com.littlek4za.booking_system.features.slot.model.TimeRange;

public record SlotResponseDto(
    Long eventId,
    Long id,
    String slotName,
    String slotDescription,
    Instant slotStartTime,
    Instant slotEndTime,
    Integer maxBookingsPerIdentity,
    Integer maxBookPerInterval,
    Integer slotIntervalMinutes,
    Integer slotFrequencyIntervalMinutes,
    Map<Integer, List<TimeRange>> businessDaysHours,
    String businessTimeZone,
    Boolean businessAllowOt,
    List<InstantRange> flexibleDaysHours,
    Instant createdAt,
    Instant updatedAt,
    Long bookingsCount
) {} 
